src/main/java/org/ehrbase/fhirbridge/ehr/converter/specific/virologischerbefund/ProAnalytZugehoerigeLaborprobeChoiceConverter.java

```diff
@@ -0,0 +1,56 @@
+package org.ehrbase.fhirbridge.ehr.converter.specific.virologischerbefund;
+
+import com.nedap.archie.rm.datavalues.DvIdentifier;
+import org.ehrbase.fhirbridge.ehr.converter.ConversionException;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.ProAnalytZugehoerigeLaborprobeChoice;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.ProAnalytZugehoerigeLaborprobeDvUri;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.ProAnalytZugehoerigeLaborprobeDvIdentifier;
+
+import org.hl7.fhir.r4.model.Observation;
+
+public class ProAnalytZugehoerigeLaborprobeChoiceConverter {
+
+    public ProAnalytZugehoerigeLaborprobeChoice convertDvIdentifier(Observation observation){
+
+        ProAnalytZugehoerigeLaborprobeDvIdentifier proAnalytZugehoerigeLaborprobeDvIdentifier = new ProAnalytZugehoerigeLaborprobeDvIdentifier();
+
+        DvIdentifier dvIdentifier = new DvIdentifier();
+
+        if(observation.getSpecimen().getIdentifier() == null){
```

Comment:
just put all of these in methods including the setters. 
So its a little bit more cleaned up. 

Identifier is 0..1 so just map it if its present.

Also not every field is required, therefore use Optional and only map what is existent. 
You have to check which fields are required for the DvIdentifier if some of them are present and others not, maybe set them with a blank field.

Reply:
Splitted up into mapping methods and boolean checker methods. Everything is also now an Optional.
https://github.com/ehrbase/fhir-bridge/pull/392/files#diff-585a1cf0f329e2ab64afa38a03d4bfe341376a4b2b2937848f1326f68da104f9